src/librustc_codegen_ssa/mir/analyze.rs

```diff
@@ -13,11 +13,12 @@ use syntax_pos::DUMMY_SP;
 use super::FunctionCx;
 use crate::traits::*;
 
-pub fn non_ssa_locals<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
+pub fn non_ssa_locals<'b, 'a: 'b, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     fx: &mut FunctionCx<'a, 'tcx, Bx>,
+    mir: &'b mut BodyCache<&'a Body<'tcx>>,
 ) -> BitSet<mir::Local> {
-    let mir = fx.mir.take().unwrap();
-    let mut analyzer = LocalAnalyzer::new(fx, mir);
+    let dominators = mir.dominators();
+    let mut analyzer = LocalAnalyzer::new(fx, mir, dominators);
```

Comment:
For some reason when trying to create the dominators within `LocalAnalyzer::new`, the fn would want to extend the lifetime of the borrow of `&mut BodyCache<...>`, which is why I moved it out here. I wasn't sure if there was a way to specify the lifetimes such that I could ensure it only retained a sharable ref on `BodyCache` and not the mutable one.

Reply:
Fixed this in the latest commit. I needed to clean up the `body` fn in `impl<'a, 'tcx> BodyCache<&'a mut Body<'tcx>>`.